lib/explorer/datasets.ex

```diff
@@ -51,8 +51,25 @@ defmodule Explorer.Datasets do
   def iris, do: read_dataset!("iris")
 
   defp read_dataset!(name) do
-    @datasets_dir
-    |> Path.join("#{name}.csv")
-    |> DataFrame.from_csv!()
+    key = {:explorer_datasets, name}
+
+    # Persistent term is used as a cache, in order to avoid
+    # several calls to the filesystem. This is mostly useful
+    # to speed up reads in tests.
+    :persistent_term.get(key, nil)
+    |> then(fn
+      nil ->
+        @datasets_dir
+        |> Path.join("#{name}.csv")
+        |> DataFrame.from_csv!()
+        |> tap(fn df ->
+          df_as_map = DataFrame.to_columns(df)
+          :persistent_term.put(key, {df.names, df_as_map})
```

Comment:
Why do you store names and columnes? Wouldn't it be better to store the thing as is?

Reply:
The problem is that we don't have the order of the columns, so I'm storing that too.
If I read the map directly, then we may have some tests failing because they rely on ordering.

Reply:
I mean, can’t you store the whole dataframe instead? :)

Reply:
@josevalim it can be! I changed in a810ffe
My previous intuition was that it could be bad for changes in the `DF` structure in the future. But I guess we don't need to worry about that.